src/picologging/streamhandler.cxx

```diff
@@ -85,11 +85,19 @@ PyObject* StreamHandler_emit(StreamHandler* self, PyObject* const* args, Py_ssiz
 }
 
 PyObject* StreamHandler_setStream(StreamHandler* self, PyObject* stream){
-    Py_XDECREF(self->stream);
+    // If stream would be unchanged, do nothing and return None
+    if (self->stream == stream) {
+        Py_RETURN_NONE;
+    }
+    // Otherwise flush current stream
+    PyObject* result = self->stream;
+    flush(self);
+    // And set new stream
     self->stream = stream;
```

Comment:
The `Py_XDECREF` needs to be added to the line before the assignment, because the reference is owned by this object (see line 97) so it counters the INCREF.

Reply:
I was confused by the fact that we're now returning it, but that makes sense. Added back.